### PR TITLE
Y24-398 - scRNA Pooling scenario restriction to study-project level

### LINKED ISSUE #4422: Y24-398 - Change pooling scenario restriction to study-project level
**User story**
Currently, there is validation on the scRNA Core submission that prevents two of the fields (number of samples per well and number of cells per chip well) varying **within a study** - added in https://github.com/sanger/sequencescape/issues/4337

R&D want to change this so that that can vary within a study, but they don't vary **within a study-project combination**.

**Who are the primary contacts for this story**
Katy, Abby

**Who is the nominated tester for UAT**
Abby

**Acceptance criteria**
To be considered successful the solution must allow:
- [ ] number of samples per well and number of cells per chip well cannot vary within a study-project combination

### PR DESCRIPTION
**Pull request by BenTopping**
Closes #4422 

#### Changes proposed in this pull request

- Updates scRNA submission templates validation for pooling restrictions to only reject on same study and project combination
- Adds 'scrna core cells per chip well' validation to ensure they are consistent in study/project combinations.

### CODE REVIEW

File: app/models/submission/validations_by_template_name.rb
[the remark this answers is not in the document]
Comment (KatyTaylor) [reply app/models/submission/validations_by_template_name.rb:1]: (link to original story - https://github.com/sanger/sequencescape/issues/4337)

File: spec/models/bulk_submission_spec.rb [hunk 1]
```diff
@@ -297,8 +297,8 @@
       it 'raises an error and sets an error message' do
         expect { subject.process }.to raise_error(ActiveRecord::RecordInvalid)
         expect(subject.errors.messages[:spreadsheet][0]).to eq(
-          "Inconsistent values for column 'scRNA Core Number of Samples per Pool' for " \
-            "Study name 'abc123_study', all rows for a specific study must have the same value"
+          "Inconsistent values for column 'scRNA Core Number of Samples per Pool' for Study name 'abc123_study' " \
+            "and Project name 'Test project', all rows for a specific study and project must have the same value"
```
Comment (KatyTaylor) [inline spec/models/bulk_submission_spec.rb:301]: You could add a new test to check the new behaviour - that it DOES allow different values for different projects within the same study.

**Review by KatyTaylor — CHANGES_REQUESTED** [review]
Thanks for doing this. Requested a couple of changes.

File: app/models/submission/validations_by_template_name.rb
[the remark this answers is not in the document]
Comment (BenTopping) [reply app/models/submission/validations_by_template_name.rb:1]: Yeah sure.

File: spec/models/bulk_submission_spec.rb (hunk 1 above)
Comment (BenTopping) [reply spec/models/bulk_submission_spec.rb:328]: I have [added a line in the valid sample sheet](https://github.com/sanger/sequencescape/pull/4437/commits/87e277e2286ef164cb9196ae4db889015b1e5179#diff-7f888b0ce579bd677f3112b85a86f2715447be6ad073940ac5ce925d93fdbdc1) to cover this case.

File: app/models/submission/validations_by_template_name.rb
[the remark this answers is not in the document]
Comment (BenTopping) [reply app/models/submission/validations_by_template_name.rb:1]: Added in [87e277e](https://github.com/sanger/sequencescape/pull/4437/commits/87e277e2286ef164cb9196ae4db889015b1e5179)

**Review by KatyTaylor — APPROVED** [review]
Thanks for making those changes.

**Pull request merged**